src/components/color-picker/color-picker-palette.tsx

```diff
@@ -60,23 +81,44 @@ export class Palette implements FormComponent {
     }
 
     private renderSwatches = () => {
-        return colors.map((color) => {
-            return brightnesses.map(this.renderSwatch(color));
-        });
+        return this.getPalette().map(this.renderSwatchButton);
     };
 
-    private renderSwatch = (color: string) => (brightness: string) => {
-        const colorName = getColorName(color, brightness);
+    private getPalette(): Swatch[] {
+        if (this.usesCustomPalette()) {
+            return (this.palette || []).map((entry) => {
+                const normalized = this.normalizeEntry(entry);
+                return {
+                    name: normalized.name || normalized.value,
+                    value: normalized.value,
+                    disabled: normalized.disabled,
+                };
+            });
+        }
+
+        // Order by brightness first, then by color.
+        return brightnesses.flatMap((b) =>
```

Comment:
This is not easy to understand what it does

Reply:
Previously, the swatches from `swatches.ts` were fed to the component like this: 
- First all the `red`s and their variations from `lightest` -> to -> `darkest`, 
- then all the `pink`s, and theri brightness variations
- so forth…

This made it hard to write a proper flexible CSS to handle the palette's grid, which could also allow consumers to use a prop like `columnCount`.

So this code now feeds the swatches to the component like this instead: 
- First all the `lightest` (of all colors), 
- then all the `light` ones, 
- then all the `default` ones, 
- etc…

This way, the CSS can only care about the columns, and not rows. And also this enables adding a prop like `paletteColumnCount`